File: tests/conformance/fixtures/typescript/24-success-criteria.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();
p.task('test')
  .run('go test ./...')
  .taskType('test')
  .successCriteria([
    { type: 'exit_code', equals: 0 },
    { type: 'file_exists', path: 'coverage.out' },
  ]);
p.task('package')
  .run('go build -o dist/app ./...')
  .taskType('package')
  .successCriteria([{ type: 'file_non_empty', path: 'dist/app' }])
  .after('test');
p.emit();

File: tests/conformance/fixtures/rust/24-success-criteria.rs
use sykli::{Pipeline, SuccessCriterion, TaskType};

fn main() {
    let mut p = Pipeline::new();
    p.task("test")
        .run("go test ./...")
        .task_type(TaskType::Test)
        .success_criteria(&[
            SuccessCriterion::ExitCode(0),
            SuccessCriterion::FileExists("coverage.out".into()),
        ]);
    p.task("package")
        .run("go build -o dist/app ./...")
        .task_type(TaskType::Package)
        .success_criteria(&[SuccessCriterion::FileNonEmpty("dist/app".into())])
        .after(&["test"]);
    p.emit();
}
